httpclient.go

```diff
@@ -97,66 +97,78 @@ func (c *HttpClientImplementation) DoRequest(req *http.Request, result interface
 	res, err := c.HttpClient.Do(req)
 	if err != nil {
 		c.Logger.Error("Cannot send request: %v", err.Error())
+		var statusCode int
+
+		if res != nil {
```

Comment:
when `res != nil` is there any chance that `res.StatusCode` nil? maybe add additional `&& res.StatusCode != nil` to be safe?

Reply:
> when `res != nil` is there any chance that `res.StatusCode` nil?

Nope, the value status code will `0`. So it is safe to check nil only on res object `res != nil`